src/plugins/jartic/archive_downloader.py

```
@@ -0,0 +1,241 @@
+                progress_bar = tqdm(
+                    total=total_size,
+                    initial=resume_pos,
+                    unit='B',
+                    unit_scale=True,
+                    unit_divisor=1024,
+                    desc=f"   {year}-{month:02d}",
+                    bar_format='{desc}: {percentage:3.0f}%|{bar:30}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}, {rate_fmt}]'
+                )
+
+                mode = 'ab' if resume_pos > 0 else 'wb'
+                with open(temp_path, mode) as f:
```

Ruff (0.12.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)